experiments/conductor/cmd/runner/utilities.go

```diff
@@ -432,3 +434,106 @@ func executeCommand(opts *RunnerOptions, cfg CommandConfig) (string, string, err
 }
+
+func checkMakeReadyPR(opts *RunnerOptions) error {
+	log.Printf("Running make ready-pr verification")
+
+	cfg := CommandConfig{
+		Name:       "Make Ready-PR",
+		Cmd:        "make",
+		Args:       []string{"ready-pr"},
+		WorkDir:    opts.branchRepoDir,
+		MaxRetries: 1,
+	}
+
+	if _, _, err := executeCommand(opts, cfg); err != nil {
+		return fmt.Errorf("make ready-pr failed: %v", err)
+	}
+
+	return nil
+}
+
+func stageChanges(ctx context.Context, opts *RunnerOptions, paths []string, commitMsg string) error {
+	// First check for changes in specified paths
+	var changedFiles []string
+	for _, path := range paths {
+		log.Printf("Checking for changes in %s", path)
+		changed, err := gitStatusCheck(opts.branchRepoDir, path)
+		if err != nil {
+			return fmt.Errorf("git status failed for %s: %v", path, err)
+		}
+		if changed {
+			changedFiles = append(changedFiles, path)
+		}
+	}
+
+	// If we have changes in specified paths, add and commit them
+	if len(changedFiles) > 0 {
+		for _, file := range changedFiles {
+			if err := gitAdd(ctx, opts.branchRepoDir, file); err != nil {
+				log.Printf("[ERROR GIT ADD]failed to stage %s: %v", file, err)
+				// best effort to continue
+			}
+		}
+		if err := gitCommit(ctx, opts.branchRepoDir, commitMsg); err != nil {
+			log.Printf("failed to commit changes: %v", err)
+		}
+	}
+
+	// Check for any other uncommitted changes
+	changed, err := gitStatusCheck(opts.branchRepoDir, ".")
+	if err != nil {
+		return fmt.Errorf("git status failed: %v", err)
+	}
+
+	if changed {
+		log.Printf("WARNING: Found unexpected changes. Trying to stage and commit them")
+		if err := gitAdd(ctx, opts.branchRepoDir, "."); err != nil {
+			return fmt.Errorf("failed to stage extra files: %v", err)
+		}
+		if err := gitCommit(ctx, opts.branchRepoDir, "ERROR: extra files modified outside expected paths"); err != nil {
+			return fmt.Errorf("failed to commit extra files: %v", err)
+		}
+		log.Printf("WARNING: Committed unexpected changes")
+	}
+
+	return nil
+}
+
+// Add this type definition
+type BranchProcessor func(ctx context.Context, opts *RunnerOptions, branch Branch) ([]string, error)
+
+func processBranches(ctx context.Context, opts *RunnerOptions, branches []Branch, processor BranchProcessor, description string, commitMsg string) {
+	for _, branch := range branches {
+		log.Printf("Processing branch %s: %s", branch.Name, description)
+
+		// Skip if branch should be skipped
+		if branch.Skip {
+			log.Printf("Skipping branch %s: marked as Skip", branch.Name)
+			continue
+		}
+
+		close := setLoggingWriter(opts, branch)
+		checkoutBranch(ctx, branch, opts.branchRepoDir)
+		// Process the branch
+		affectedPaths, err := processor(ctx, opts, branch)
+		if err != nil {
+			log.Printf("failed to process branch %s: %v", branch.Name, err)
+			close()
+			continue
+		}
+
+		// Run make ready-pr verification
+		if err := checkMakeReadyPR(opts); err != nil {
```

Comment:
I'm still not convinced that we should run "make ready-pr" after every step. First, the code might not compile at certain command. For example, after generating the types but before running the generate-crds script to create the deepcopy file. Second, doing so could add a lot of unnecessary overhead.

Reply:
We could instead run "go fmt" after each step.

Reply:
added a runLinters for each step. Currently we run go fmt. We can add to it.